### PR TITLE
TaskThread: avoid handling exceptions from task_computed with _fail

### PR DESCRIPTION
**Pull request by etam**
Fixes: #2671

### CODE REVIEW

File: apps/blender/task/blenderrendertask.py [hunk 1]
```diff
@@ -391,8 +392,10 @@ def initialize(self, dir_manager):
 
     @coretask.accepting
     # pylint: disable-msg=too-many-locals
-    def query_extra_data(self, perf_index, num_cores=0, node_id=None,
-                         node_name=None):
+    def query_extra_data(self, perf_index: float, num_cores: int = 0,
+                         node_id: Optional[str] = None,
+                         node_name: Optional[str] = None) \
+            -> FrameRenderingTask.ExtraData:
```
Comment (tworec) [inline apps/blender/task/blenderrendertask.py:398]: 👍 cool, that you're adding typing

File: apps/core/benchmark/benchmarkrunner.py [hunk 2]
```diff
@@ -52,8 +54,8 @@ def _get_task_thread(self, ctd):
             raise Exception("No docker container found")
         return super(BenchmarkRunner, self)._get_task_thread(ctd)
 
-    def is_success(self, task_thread):
-        if not task_thread.result:
+    def is_success(self, task_thread: TaskThread) -> bool:
+        if task_thread.error or not task_thread.result:
```
Comment (tworec) [inline apps/core/benchmark/benchmarkrunner.py:58]: is this the main fix here?

File: golem/task/localcomputer.py [hunk 3]
```diff
@@ -88,23 +91,26 @@ def get_progress(self):
...
         self.end_time = time.time()
         if self.is_success(task_thread):
             self.computation_success(task_thread)
         else:
             self.computation_failure(task_thread)
 
-    def is_success(self, task_thread):
+    # This cannot be changed to staticmethod, because it's overriden in
+    # a derived class
+    # pylint:disable=no-self-use
```
Comment (tworec) [inline golem/task/localcomputer.py:103]: 👍 for comment

File: tests/apps/core/benchmark/test_benchmarkrunner.py [hunk 4]
```diff
@@ -6,20 +6,70 @@
 from golem.testutils import TempDirFixture
 
 
-class BenchmarkRunnerTest(TempDirFixture):
+class DummyTask(Task):
```
Comment (tworec) [inline tests/apps/core/benchmark/test_benchmarkrunner.py:9]: what is this empty class for?

File: tests/apps/core/benchmark/test_benchmarkrunner.py [hunk 5]
```diff
@@ -77,37 +141,44 @@ def test_task_computed_immidiately(self):
...
-        # False result and False error_msg
         task_thread.result = None
+        task_thread.error = False
         task_thread.error_msg = error_msg = None
         self.instance.error_callback = error_mock = mock.MagicMock()
         self.instance.task_computed(task_thread)
         error_mock.assert_called_once_with(error_msg)
 
-        # False result and Non-False error_msg
+    def test_task_computed_false_resulst_and_non_false_error_msg(self):
```
Comment (tworec) [inline tests/apps/core/benchmark/test_benchmarkrunner.py:153]: :+1: veeery goood to split long tests

**Review by tworec — COMMENTED** [review]
lgtm. will accept when you answer my q

File: tests/apps/core/benchmark/test_benchmarkrunner.py (hunk 4 above)
Comment (etam) [reply tests/apps/core/benchmark/test_benchmarkrunner.py:9]: `Task` derives from `abc.ABC` and the methods below are marked as `@abstractmethod`. You can't create objects of it. In tests this was worked around by u sing `@mock.patch.multiple(Task, __abstractmethods__=frozenset())`. It was ugly and was making me problems when refactoring tests around `WrongTask`, which derives from `Task`.

File: apps/core/benchmark/benchmarkrunner.py (hunk 2 above)
Comment (etam) [reply apps/core/benchmark/benchmarkrunner.py:58]: No. The main fix is in adding `else` part after `try` and `except` in `DockerTaskThread.run`.

**Review by tworec — APPROVED** [review]

**Comment by tworec** [comment]
thx for answers

**Pull request merged**